loginScreen/loginScreen/New Group/Controllers/LoginViewController.swift

```diff
@@ -0,0 +1,394 @@
+        let button = UIButton()
+        button.layer.cornerRadius = preferredSubviewsHeight / 2
+        button.createShadowLayer(withCornerradius: preferredSubviewsHeight / 2)
+        button.backgroundColor = UIColor(red: 0.33, green: 0.42, blue: 0.71, alpha: 1.00)
+
+        let image = viewWithAttachedImage(withSize: CGSize(width: preferredSubviewsHeight * 1.4,
+                                                           height: preferredSubviewsHeight),
+                                          scale: 0.32,
+                                          imageNamed: "twitter.png")
+        
+        let title = UILabel()
+        title.text = "Twitter"
+        title.textColor = .white
+        
+        button.addSubview(image)
+        button.addSubview(title)
+        button.addTarget(self, action: #selector(anyAction), for: .touchUpInside)
+
+        return button
+    }()
+    
+    private lazy var separatorView: UIView = {
+        let attached = UIView()
+        
+        let leftSeparator = UIView()
+        let rigtSeparator = UIView()
+        leftSeparator.backgroundColor = .lightGray
+        rigtSeparator.backgroundColor = .lightGray
+        
+        let centerLabel = UILabel()
+        centerLabel.text = "or connect with"
+        centerLabel.textAlignment = .center
+        centerLabel.textColor = .lightGray
+        
+        attached.addSubview(leftSeparator)
+        attached.addSubview(centerLabel)
+        attached.addSubview(rigtSeparator)
+        
+        return attached
+    }()
+    
+    // MARK: - Lifecycle
+    
+    override func viewDidLoad() {
+        super.viewDidLoad()
+        setupView()
+        setupHierarchy()
+        setupTopStackLayout()
+        setupBottomStackLayout()
+    }
+    
+    // MARK: - Setup view
+    
+    override var preferredStatusBarStyle: UIStatusBarStyle {
+        return .lightContent
+    }
+    
+    private func setupView() {
+        if let background = UIImage(named: "background.jpg") {
+            let imageView = UIImageView(frame: view.bounds)
+            imageView.contentMode = UIView.ContentMode.scaleAspectFill
+            imageView.clipsToBounds = true
+            imageView.image = background
+            imageView.center = view.center
+            view.addSubview(imageView)
+        } else {
+            view.backgroundColor = UIColor(red: 0.44, green: 0.41, blue: 0.82, alpha: 1.00)
+        }
+    }
+    
+    // MARK: - Setup hierarchy
+    
+    private func setupHierarchy() {
+        view.addSubview(topStack)
+        view.addSubview(bottomStack)
+    }
+    
+    // MARK: - Setup top stack layout
+    
+    private func setupTopStackLayout() {
+        topStack.snp.makeConstraints { make in
+            make.top.equalTo(view.safeAreaLayoutGuide.snp.top)
+            make.width.equalTo(view).multipliedBy(0.8)
+            make.centerX.equalTo(view)
+            make.bottom.equalTo(view.safeAreaLayoutGuide.snp.centerY)
+        }
```

Comment:
Можно проще. Остальное по аналогии
```suggestion
        topStack.snp.makeConstraints {
            $0.top.equalTo(view.safeAreaLayoutGuide.snp.top)
            $0.width.equalToSuperview().multipliedBy(0.8)
            $0.centerX.equalToSuperview()
            $0.bottom.equalTo(view.safeAreaLayoutGuide.snp.centerY)
        }
```